src/pages/[platform]/build-a-backend/auth/concepts/multi-factor-authentication/index.mdx

```diff
@@ -44,31 +48,160 @@ export const auth = defineAuth({
+| **Description** | Generates a short-lived numeric code for user authentication that includes a shared secret key and current time using an authenticator app. | Generates a one-time code shared via text message that is entered with other credentials for user authentication. | Generates a one-time code sent to the user's registered email address. The user must access the email and enter the code to complete the authentication process. |
+| **Benefits** | More secure than SMS since the code is generated locally and not transmitted over a network. TOTP also works without cell service as long as the TOTP app is installed. | Easy to set up with a user-provided phone number and is familiar to users as a common authentication method. | Email is a widely used and familiar communication channel that requires no additional hardware or software requirements on the user's end. |
+| **Constraints** | Requires an app to generate codes and adds to the initial setup of an account. Codes also expire quickly and must be used promptly after it is generated. | SMS requires cell service and can include an additional cost for the user. Although rare, SMS messages can also be intercepted. | Depends on the availability and reliability of email services. Although rare, emails can be intercepted or accounts can become compromised. |
 
 </Accordion>
 
+
+<InlineFilter filters={["angular", "javascript", "nextjs", "react", "vue", "android"]}>
+If multiple MFA methods are enabled for the user, the `signIn` API will return `CONTINUE_SIGN_IN_WITH_MFA_SELECTION` as the next step in the auth flow. During this scenario, the user should be prompted to select the MFA method they want to use to sign in and their preference should be passed to `confirmSignIn`.
+
+```ts
+import { confirmSignIn, SignInOutput } from 'aws-amplify/auth';
+
+function handleSignInNextSteps(output: SignInOutput) {
+  const { nextStep } = output;
+  switch (nextStep.signInStep) {
+    // ...
+    case 'CONTINUE_SIGN_IN_WITH_MFA_SELECTION':
+      const allowedMFATypes = nextStep.allowedMFATypes;
+      const mfaType = promptUserForMFAType(allowedMFATypes);
+    case 'CONFIRM_SIGN_IN_WITH_SMS_CODE':
+      // prompt user to enter otp code delivered via SMS
+      break;
+    case 'CONFIRM_SIGN_IN_WITH_TOTP_CODE':
+      // prompt user to enter otp code from their authenticator app
+      break;
+    case 'CONFIRM_SIGN_IN_WITH_EMAIL_CODE':
+      // prompt user to enter otp code delivered via EMAIL
+      break;
+    // ...
+  }
+}
+
+function promptUserForMFAType(
+  allowedMFATypes?: ('SMS' | 'TOTP' | 'EMAIL')[]
+): 'SMS' | 'TOTP' | 'EMAIL' {
+  // Prompt user to select MFA type
+}
+
+async function handleMFASelection(mfaType: 'SMS' | 'TOTP' | 'EMAIL') {
+  try {
+    const output = await confirmSignIn({
+      challengeResponse: mfaType
+    });
+    handleSignInNextSteps(output);
+  } catch (error) {
+    console.log(error);
+  }
+}
+```
+</InlineFilter>
+
+<InlineFilter filters={["swift", "flutter"]}>
```

Comment:
is this callout also true for android?

Reply:
Yes, this would also be true for android. I believe the enum value is the same as JS so would be covered by that callout but we are missing a code example. 

@vincetran would you be able to confirm and spin up a quick code example for `CONTINUE_SIGN_IN_WITH_MFA_SELECTION`?

Reply:
Added!